scala-api/src/main/scala/com/cloudera/livy/ScalaJobHandle.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cloudera.livy
+
+import com.cloudera.livy.JobHandle.State
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{CanAwait, ExecutionContext, Future, TimeoutException}
+import scala.util.Try
+
+class ScalaJobHandle[T](jobHandle: JobHandle[T]) extends Future[T] {
```

Comment:
It would be nice to have a unit test suite for this class, to make sure you're implementing the methods correctly. You can use mockito to mock a Java future, to make testing easier.